src/views/Header/HeaderBackButton.tsx

```diff
@@ -2,180 +2,222 @@ import * as React from 'react';
-            !!tintColor && { tintColor },
+            Boolean(labelVisible) && styles.iconWithLabel,
+            Boolean(tintColor) && { tintColor },
           ]}
-          source={defaultBackImage}
+          source={require('../assets/back-icon.png')}
           fadeDuration={0}
         />
       );
     }
-  }
-
-  private getTitleText = () => {
-    const { width, title, truncatedTitle } = this.props;
-
-    let { initialTextWidth } = this.state;
-
-    if (title === null) {
-      return null;
-    } else if (!title) {
-      return truncatedTitle;
-    } else if (initialTextWidth && width && initialTextWidth > width) {
-      return truncatedTitle;
-    } else {
-      return title;
-    }
   };
 
-  private maybeRenderTitle() {
+  private renderLabel() {
     const {
+      label,
+      truncatedLabel,
       allowFontScaling,
-      backTitleVisible,
-      titleStyle,
+      labelVisible,
+      backImage,
+      labelStyle,
       tintColor,
+      screenLayout,
     } = this.props;
-    let backTitleText = this.getTitleText();
 
-    if (!backTitleVisible || backTitleText === null) {
+    const leftLabelText = this.shouldTruncateLabel() ? truncatedLabel : label;
+
+    if (!labelVisible || leftLabelText === undefined) {
       return null;
     }
 
-    return (
-      <Text
+    const labelElement = (
+      <Animated.Text
         accessible={false}
-        onLayout={this.handleTextLayout}
-        style={[styles.title, !!tintColor && { color: tintColor }, titleStyle]}
+        onLayout={this.handleLabelLayout}
+        style={[
+          styles.label,
+          tintColor ? { color: tintColor } : null,
+          labelStyle,
+        ]}
         numberOfLines={1}
         allowFontScaling={!!allowFontScaling}
       >
-        {this.getTitleText()}
-      </Text>
+        {leftLabelText}
+      </Animated.Text>
+    );
+
+    if (backImage || Platform.OS !== 'ios') {
+      // When a custom backimage is specified, we can't mask the label
+      // Otherwise there might be weird effect due to our mask not being the same as the image
+      return labelElement;
+    }
+
+    return (
+      <MaskedViewIOS
+        maskElement={
+          <View style={styles.iconMaskContainer}>
+            <Image
+              source={require('../assets/back-icon-mask.png')}
+              style={styles.iconMask}
+            />
+            <View style={styles.iconMaskFillerRect} />
+          </View>
+        }
+      >
+        <View
+          style={
+            screenLayout
+              ? // We make the button extend till the middle of the screen
+                // Otherwise it appears to cut off when translating
+                [styles.labelWrapper, { minWidth: screenLayout.width / 2 - 27 }]
+              : null
+          }
+        >
+          {labelElement}
+        </View>
+      </MaskedViewIOS>
     );
   }
 
+  private handlePress = () =>
+    this.props.onPress && requestAnimationFrame(this.props.onPress);
```

Comment:
why?

Reply:
when `onPress` is called, the touchable opacity animation waits till this function returns. since we do expensive work re-rendering stuff, it can feel laggy. doing the task asynchronously doesn't block the touchable animation so will feel more responsive